cpu/stm32l1/periph/i2c.c

```diff
@@ -66,13 +63,12 @@ static mutex_t locks[] =  {
 
 int i2c_init_master(i2c_t dev, i2c_speed_t speed)
 {
-    I2C_TypeDef *i2c;
-    GPIO_TypeDef *port;
-    int pin_scl = 0, pin_sda = 0;
+    I2C_TypeDef *i2c = i2c_config[dev].dev;
```

Comment:
@ReneHerthel please put this check in again. We already discussed it some comments ago.